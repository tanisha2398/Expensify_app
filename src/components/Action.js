import React from "react";
const Action = props => (
  <div>
    <button onClick={props.handlePick} disabled={!props.hasOptions}>
      What Should I do??
    </button>
  </div>
);
export default Action;
