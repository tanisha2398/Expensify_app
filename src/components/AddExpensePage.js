import React from "react";
const AddExpensePage = () => <div>this from add expense component</div>;
export default AddExpensePage;
