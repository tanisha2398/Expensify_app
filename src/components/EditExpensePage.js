import React from "react";
const EditExpensePage = () => <div>this from EditExpensePage </div>;
export default EditExpensePage;
