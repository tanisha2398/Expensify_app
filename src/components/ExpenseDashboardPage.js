import React from "react";

const ExpenseDashboardPage = () => <div>this from dashboard component</div>;
export default ExpenseDashboardPage;
