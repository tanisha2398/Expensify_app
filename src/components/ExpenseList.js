import React from "react";
import { connect } from "react-redux";
import ExpenseItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";
const ExpenseList = props => (
  <div>
    <h1>expense list</h1>
    {props.expenses.map(expense => (
      <ExpenseItem key={expense.id} {...expense} />
    ))}
  </div>
);
const mapStateToProps = state => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};
export default connect(mapStateToProps)(ExpenseList);
