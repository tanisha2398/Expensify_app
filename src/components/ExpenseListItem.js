import React from "react";
import moment from "moment";
import numeral from "numeral";
import { Link } from "react-router-dom";
export const ExpenseItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {numeral(amount / 100).format("$0,0.00")}-
      {moment(createdAt).format("MMMM Do, YYYY")}
    </p>
  </div>
);

export default ExpenseItem;
