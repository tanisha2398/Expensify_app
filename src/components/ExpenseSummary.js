import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import getExpenseTotal from "../selectors/expenses-total";

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(expensesTotal / 100).format("$0,0.00");
  return (
    <div>
      <p>
        Viewing {expenseCount} {expenseWord} totalling
        {formattedExpensesTotal}
      </p>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleExpense = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpense.length,
    expensesTotal: getExpenseTotal(visibleExpense)
  };
};
export default connect(mapStateToProps)(ExpenseSummary);
