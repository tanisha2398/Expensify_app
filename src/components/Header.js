import React from "react";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";
import { Link } from "react-router-dom";
export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/dashboard">
          <h1>Expensify</h1>
        </Link>

        <button onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(Header);
