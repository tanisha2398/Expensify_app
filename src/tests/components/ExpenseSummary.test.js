import React from "react";
import { shallow } from "enzyme";
import { ExpenseSummary } from "../../components/ExpenseSummary";

test("should correctly render ExpenseSummary with 1 expense", () => {
  const wrapper = shallow(
    <ExpenseSummary expenseCount={1} expensesTotal={234} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpenseSummary with multiple expense", () => {
  const wrapper = shallow(
    <ExpenseSummary expenseCount={23} expensesTotal={23524321523} />
  );
  expect(wrapper).toMatchSnapshot();
});
