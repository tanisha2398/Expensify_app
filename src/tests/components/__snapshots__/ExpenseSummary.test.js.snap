// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <p>
    Viewing 
     
    expenses
     totalling
    $0.00
  </p>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expense 1`] = `
<div>
  <p>
    Viewing 
     
    expenses
     totalling
    $0.00
  </p>
</div>
`;
